File: AllSpiceSharp/Models/Recipe.cs
namespace AllSpiceSharp.Models;

public class Recipe : IRepoItem<int>
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string Title { get; set; }
  public string Instructions { get; set; }
  public string Img { get; set; }
  public string category { get; set; }
  public string creatorId { get; set; }
  public Profile Creator { get; set; }
}

File: AllSpiceSharp.client/src/models/Recipe.js
export class Recipe {
  constructor(data) {
    this.id = data.id
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.title = data.title
    this.instructions = data.instructions
    this.img = data.img
    this.category = data.category
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.favoriteCount = data.favoriteCount
    this.favorited = data.favorited
  }
}